core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -48,6 +51,31 @@ private[spark] object JettyUtils extends Logging {
   val SPARK_CONNECTOR_NAME = "Spark"
   val REDIRECT_CONNECTOR_NAME = "HttpsRedirect"
 
+  val snappyDataRealm = "SnappyDataPulse"
+  val snappyDataRoles = Array("user")
+  var customAuthenticator: Option[BasicAuthenticator] = None
+
+  val constraintMapping = {
+    val constraint = new Constraint()
+    constraint.setName(Constraint.__BASIC_AUTH);
+    constraint.setRoles(snappyDataRoles);
+    constraint.setAuthenticate(true);
+
+    val cm = new ConstraintMapping();
+    cm.setConstraint(constraint);
+    cm.setPathSpec("/*")
+    cm
+  }
+
+  val snappyHashLoginService = {
```

Comment:
Can make it lazy val if same instance can be passed for each of the ServletContextHandlers.